pkg/storage/registrator/agent/agent.go

```diff
@@ -30,9 +31,9 @@ func GetProperty(props map[string]interface{}, t taxonomy.ConnectionType, key st
 	}
 	switch propertyMap := propertyMap.(type) {
 	case map[string]interface{}:
-		property := propertyMap[key]
-		if property != nil {
-			return property.(string), nil
+		property, found := propertyMap[key]
+		if found && (property != nil) {
```

Comment:
why do you separate `nil` return and unfound, if you have the same processing ?
why not just `if property != nil ` ?

Reply:
changed